tabs/pid_tuning.js

```diff
@@ -701,11 +718,8 @@ TABS.pid_tuning.initRatesPreview = function () {
     this.keepRendering = true;
     this.model = new Model($('.rates_preview'), $('.rates_preview canvas'));
 
-    var scale = d3.scale.linear().domain([900, 2100]);
-
-    this.rollScale = scale.range([Math.PI * 2, -Math.PI * 2]);
-    this.pitchScale = scale.range([Math.PI * 2, -Math.PI * 2]);
-    this.yawScale = scale.range([Math.PI * 2, -Math.PI * 2]);
+    this.$superExpo = $('.pid_tuning input[name="show_superexpo_rates"]');
```

Comment:
I also miss a good IDE for this stuff. I usually just do c or c++ over the
last 2 years and really got spoiled by easeness of a good IDE where I
easily can browse through different files.
Op 7 jul. 2016 22:03 schreef "Michael Keller" <notifications@github.com>:

> In tabs/pid_tuning.js
> <https://github.com/betaflight/betaflight-configurator/pull/89#discussion_r69976533>
> :
>
> > @@ -701,11 +718,8 @@ TABS.pid_tuning.initRatesPreview = function () {
> >      this.keepRendering = true;
> >      this.model = new Model($('.rates_preview'), $('.rates_preview canvas'));
> >
> > -    var scale = d3.scale.linear().domain([900, 2100]);
> > -
> > -    this.rollScale = scale.range([Math.PI * 2, -Math.PI * 2]);
> > -    this.pitchScale = scale.range([Math.PI * 2, -Math.PI * 2]);
> > -    this.yawScale = scale.range([Math.PI * 2, -Math.PI * 2]);
> > +    this.$superExpo = $('.pid_tuning input[name="show_superexpo_rates"]');
>
> Well without code reviews pretty much anything is a recipe for disaster.
>
> But this is a very new project, and it was started with the intention of
> making the 'extra features' that Betaflight has over cleanflight accessible
> from the GUI. Once this has been achieved (and I think we are close), I
> would like to add more structure to it and introduce a set of coding style
> rules and a review process. But ultimately, @borisbstyle
> <https://github.com/borisbstyle> is the owner of this project and he will
> have to agree with this course of action.
>
> I'm sorry, but I keep noticing that you sound a bit like "let's not change
> anything but stick to the current mess,
>
> This is not quite the case. However, I believe that cleaning up a codebase
> has to start with specifying how 'clean code' has to look. Before this is
> done, introducing new paradigms into the codebase does not do anything to
> clean it up, it just makes it harder to clean up if these paradigms do not
> become part of the spec. As I said above, I think we are close to a point
> where we should be shifting our focus to add more structure, because the
> user requirements are almost met...
>
> I'm used to snake_case for variables and camelCase for methods (which is
> the case in many JS projects)
>
> I have never come across this one. Doesn't this make the code a bit hard
> to read? For me it's not so much about personal preferences as it is about
> having a common set of rules and every contributor sticking to it.
>
> if (self.currentRates.hasOwnProperty(targetElement.attr('name')) && targetValue) {
>   self.currentRates[targetElement.attr('name')] = targetValue * 100;
> }
>
> I saw that. I am personally not a huge fan of this sort of 'optimisation',
> as it tends to break on changes to functionality. Case in point is the code
> that I've had to add in order to make it not break for non-current versions
> of the firmware:
>
>  if (targetElement.attr('name') === 'rc_rate' && CONFIG.flightControllerIdentifier !== "BTFL" || semver.lt(CONFIG.flightControllerVersion, "2.8.1")) {
>      self.currentRates.rc_rate_yaw = targetValue * 100;
> }
>
> if (targetElement.attr('name') === 'roll_pitch_rate' && semver.lt(CONFIG.apiVersion, "1.7.0")) {
>     self.currentRates.roll_rate = targetValue * 100;
>     self.currentRates.pitch_rate = targetValue * 100;
> }
>
> I agree that the Chrome console is pretty neat for debugging. What I am
> looking for is a tool that gives me some sort of 'code intelligence', i.e.
> can reliably find references to objects, and provides assistance for
> refactoring. I have had some success for this in JS with WebStorm in the
> past, but this does not come in a free version.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/betaflight/betaflight-configurator/pull/89/files/86352b2a61bc2e307fb735ee899be57653d5cd9b#r69976533>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AKQlhIYUwQ9RKzSRDVHivRkPGYqi8moRks5qTVuZgaJpZM4JGjsR>
> .
>